tests/ui/simd/repr_packed.rs

```diff
@@ -6,9 +6,6 @@
 #[repr(simd, packed)]
 struct Simd<T, const N: usize>([T; N]);
 
-#[repr(simd)]
```

Comment:
I think there is a non-zero possibility that we end up with both types--one packed and one not. Notably, adding `repr(packed)` changes the (internal rustc) ABI from "vector" to "aggregate" and involves an extra load.  I could see a situation where someone doesn't mind the padding and wants the (chance of) slightly better codegen opportunity.

It's of course slightly odd that it's not byte-aligned, but `repr(packed(N))` does take an alignment argument, with `repr(packed, simd)` it's just a different default alignment.

Reply:
right. as far as I am concerned what we're really doing here is altering a very-poorly-defined lang item, since `repr(simd)` and `repr(simd, packed)` are Basically Magic that people can't really use directly, and what we want to think about is how to describe something that we can then expose to people.